File: kaizen-openapi-parser/src/main/java/com/reprezen/kaizen/oasparser/model3/OAuthFlow.java
package com.reprezen.kaizen.oasparser.model3;

import java.util.Map;

import javax.annotation.Generated;

import com.reprezen.jsonoverlay.IJsonOverlay;
import com.reprezen.jsonoverlay.IModelPart;

public interface OAuthFlow extends IJsonOverlay<OAuthFlow>, IModelPart<OpenApi3, OAuthFlow> {

	// AuthorizationUrl
	@Generated("com.reprezen.gen.CodeGenerator")
	String getAuthorizationUrl();

	@Generated("com.reprezen.gen.CodeGenerator")
	String getAuthorizationUrl(boolean elaborate);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setAuthorizationUrl(String authorizationUrl);

	// TokenUrl
	@Generated("com.reprezen.gen.CodeGenerator")
	String getTokenUrl();

	@Generated("com.reprezen.gen.CodeGenerator")
	String getTokenUrl(boolean elaborate);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setTokenUrl(String tokenUrl);

	// RefreshUrl
	@Generated("com.reprezen.gen.CodeGenerator")
	String getRefreshUrl();

	@Generated("com.reprezen.gen.CodeGenerator")
	String getRefreshUrl(boolean elaborate);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setRefreshUrl(String refreshUrl);

	// Scope
	@Generated("com.reprezen.gen.CodeGenerator")
	Map<String, String> getScopes();

	@Generated("com.reprezen.gen.CodeGenerator")
	Map<String, String> getScopes(boolean elaborate);

	@Generated("com.reprezen.gen.CodeGenerator")
	boolean hasScope(String name);

	@Generated("com.reprezen.gen.CodeGenerator")
	String getScope(String name);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setScopes(Map<String, String> scopes);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setScope(String name, String scope);

	@Generated("com.reprezen.gen.CodeGenerator")
	void removeScope(String name);

	// ScopesExtension
	@Generated("com.reprezen.gen.CodeGenerator")
	Map<String, Object> getScopesExtensions();

	@Generated("com.reprezen.gen.CodeGenerator")
	Map<String, Object> getScopesExtensions(boolean elaborate);

	@Generated("com.reprezen.gen.CodeGenerator")
	boolean hasScopesExtension(String name);

	@Generated("com.reprezen.gen.CodeGenerator")
	Object getScopesExtension(String name);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setScopesExtensions(Map<String, Object> scopesExtensions);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setScopesExtension(String name, Object scopesExtension);

	@Generated("com.reprezen.gen.CodeGenerator")
	void removeScopesExtension(String name);

	// Extension
	@Generated("com.reprezen.gen.CodeGenerator")
	Map<String, Object> getExtensions();

	@Generated("com.reprezen.gen.CodeGenerator")
	Map<String, Object> getExtensions(boolean elaborate);

	@Generated("com.reprezen.gen.CodeGenerator")
	boolean hasExtension(String name);

	@Generated("com.reprezen.gen.CodeGenerator")
	Object getExtension(String name);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setExtensions(Map<String, Object> extensions);

	@Generated("com.reprezen.gen.CodeGenerator")
	void setExtension(String name, Object extension);

	@Generated("com.reprezen.gen.CodeGenerator")
	void removeExtension(String name);
}

File: parser/src/main/kotlin/com/reprezen/kaizen/oasparser/model3/OAuthFlow.kt
package com.reprezen.kaizen.oasparser.model3

import com.reprezen.jsonoverlay.parser.Generated
import com.reprezen.jsonoverlay.IJsonOverlay
import com.reprezen.jsonoverlay.IModelPart
import kotlin.collections.Map

interface OAuthFlow : IJsonOverlay<OAuthFlow>, IModelPart<OpenApi3, OAuthFlow> {

	// AuthorizationUrl
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getAuthorizationUrl() : String?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setAuthorizationUrl(authorizationUrl : String)

	// TokenUrl
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getTokenUrl() : String?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setTokenUrl(tokenUrl : String)

	// RefreshUrl
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getRefreshUrl() : String?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setRefreshUrl(refreshUrl : String)

	// Scope
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getScopes() : MutableMap<String, String>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getScopes(elaborate : Boolean) : MutableMap<String, String>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasScopes() : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasScope(name : String) : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getScope(name : String) : String?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setScopes(scopes : MutableMap<String, String>)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setScope(name : String,scope : String)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun removeScope(name : String)

	// ScopesExtension
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getScopesExtensions() : MutableMap<String, Any>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getScopesExtensions(elaborate : Boolean) : MutableMap<String, Any>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasScopesExtensions() : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasScopesExtension(name : String) : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getScopesExtension(name : String) : Any?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setScopesExtensions(scopesExtensions : MutableMap<String, Any>)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setScopesExtension(name : String,scopesExtension : Any)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun removeScopesExtension(name : String)

	// Extension
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getExtensions() : MutableMap<String, Any>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getExtensions(elaborate : Boolean) : MutableMap<String, Any>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasExtensions() : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasExtension(name : String) : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getExtension(name : String) : Any?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setExtensions(extensions : MutableMap<String, Any>)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setExtension(name : String,extension : Any)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun removeExtension(name : String)

}
